lib/logstash/filters/grok.rb

```diff
@@ -215,6 +219,12 @@ class LogStash::Filters::Grok < LogStash::Filters::Base
     # Set to 0 to disable timeouts
     config :timeout_millis, :validate => :number, :default => 30000
 
+    # Group timeout for multiple patterns instead of having a timeout per pattern.
+    # When set to true timeout_millis effectively becomes a timeout over several patterns.
+    # Default is false due backwards compatibility.
+    # Has only an effect when timeout_millis > 0
+    config :timeout_grouped, :validate => :boolean, :default => false
```

Comment:
maybe `:grouped_timeout` or `:global_timeout` ?

Reply:
What about `timeout_scope => event` and `timeout_scope => pattern`? I think that this is a little more clear, and also gives us the freedom to find alternate solutions later on.


```suggestion
    config :timeout_scope, :validate => %w(pattern event), :default => "pattern"
```

Reply:
like them all :) ... leaning towards `global_timeout` (or `shared_timeout`) for now.
`timeout_scope` would make sense here but if we think forward will it make sense to use the same option in **kv** plugin? if so I am okay with that.

Reply:
although on the other hand `grouped_timeout` should evoke that its "grouping" multiple patterns.
I am obviously biased, some long-term (native) speaker and LS user will need to decide for me.
// cc @karenzone do you have thoughts for us? what would sounds natural to you?

Reply:
I think the reason that a Boolean option feels wrong to me is that it doesn't clearly indicate the _level_ of grouping, and with Logstash pipelines we already deal with different forms of grouping (e.g., batched processing of events).

By making the user specify the scope as a bareword value that clearly indicates the desired behaviour, we enable them to understand it at a glance and we remove some of the implicit context.

Reply:
@yaauie thanks for the insight - that nails it for me, will change to `timeout_scope`